### ISSUE TITLE
proxy-injector produces invalid patch in GKE 10.7 cluster

### ISSUE THREAD

**Issue by klingerf** [linkerd/linkerd2#1756]
### What is the issue?

In a GKE 10.7 cluster, the admission controller that's installed when installing linkerd with the `--proxy-auto-inject` flag set produces a patch to a pod template spec that's invalid, causing the apply to fail, with:

```
Error from server (Invalid): error when creating "STDIN": Deployment.apps "emoji" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"linkerd.io/control-plane-ns":"linkerd", "linkerd.io/proxy-deployment":"emoji"}: `selector` does not match template `labels`
```

### How can it be reproduced?

This can be reproduced on a GKE 10.7 cluster, using the emojivoto app.

1. Install linkerd version `edge-18.10.2`:

```
linkerd install --tls optional --proxy-auto-inject | kubectl apply -f -
```

2. Wait for it to start

```
linkerd check
```

3. Install the emojivoto app, which fails with:

```
curl -s https://run.linkerd.io/emojivoto.yml | kubectl apply -f -
namespace "emojivoto" configured
service "emoji-svc" unchanged
service "voting-svc" unchanged
service "web-svc" unchanged
Error from server (Invalid): error when creating "STDIN": Deployment.apps "emoji" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"linkerd.io/control-plane-ns":"linkerd", "linkerd.io/proxy-deployment":"emoji"}: `selector` does not match template `labels`
Error from server (Invalid): error when creating "STDIN": Deployment.apps "voting" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"linkerd.io/control-plane-ns":"linkerd", "linkerd.io/proxy-deployment":"voting"}: `selector` does not match template `labels`
Error from server (Invalid): error when creating "STDIN": Deployment.apps "web" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"linkerd.io/control-plane-ns":"linkerd", "linkerd.io/proxy-deployment":"web"}: `selector` does not match template `labels`
Error from server (Invalid): error when creating "STDIN": Deployment.apps "vote-bot" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"linkerd.io/proxy-deployment":"vote-bot", "linkerd.io/control-plane-ns":"linkerd"}: `selector` does not match template `labels`
```

### Environment

- Kubernetes Version:

```
$ kubectl version --short
Client Version: v1.10.7
Server Version: v1.10.7-gke.6
```

- Cluster Environment: GKE

- Host OS:

```
System Info:
 Machine ID:                 d7291c2c486553cf22a285b1d643d0f5
 System UUID:                D7291C2C-4865-53CF-22A2-85B1D643D0F5
 Boot ID:                    1450cd74-7d57-4acc-9489-fe7c6544863b
 Kernel Version:             4.14.65+
 OS Image:                   Container-Optimized OS from Google
 Operating System:           linux
 Architecture:               amd64
 Container Runtime Version:  docker://17.3.2
 Kubelet Version:            v1.10.7-gke.6
 Kube-Proxy Version:         v1.10.7-gke.6
```

- Linkerd version:

```
$ linkerd version
Client version: edge-18.10.2
Server version: edge-18.10.2
```

### Additional context

Note that proxy injection works as expected when testing on a 10.3 Docker for Mac cluster.

I think the issue is that newer versions of kubernetes enforce this constraint on selectors:

> `.spec.selector` must match `.spec.template.metadata.labels`, or it will be rejected by the API.

Relates to kubernetes/kubernetes#26202.
Relates to helm/helm#2494.

**Comment by ihcsim** [comment]
@klingerf Can confirm that this is reproducible in on GKE 1.10.7. After some more testing (on GKE 1.10.7) with the pod template labels and annotations, I found a bug in the proxy injector where the webhook's will overwrite any existing labels and annotations while mutating the pod template spec 😲 

I will be submitting a PR with the fixes.

**Comment by klingerf** [comment]
Great, thanks for tracking this down, @ihcsim! I'll review #1763 and #1765 today.

**Comment by klingerf** [comment]
This was fixed by #1765.
